cmake/sca/eclair/sca.cmake

```diff
@@ -0,0 +1,140 @@
+endif()
+if (ECLAIR_SUMMARY_TXT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-summary_txt=${ECLAIR_OUTPUT_DIR}/summary_txt")
+endif()
+if (ECLAIR_SUMMARY_DOC)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-summary_doc=${ECLAIR_OUTPUT_DIR}/summary_doc")
+endif()
+if (ECLAIR_SUMMARY_ODT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-summary_odt=${ECLAIR_OUTPUT_DIR}/summary_odt")
+endif()
+if (ECLAIR_FULL_TXT_ALL_AREAS)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,areas")
+endif()
+if (ECLAIR_FULL_TXT_FIRST_AREA)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,first_area")
+endif()
+if (ECLAIR_FULL_TXT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-full_txt=${ECLAIR_OUTPUT_DIR}/report_full_txt")
+endif()
+if (ECLAIR_FULL_DOC_ALL_AREAS)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,areas")
+endif()
+if (ECLAIR_FULL_DOC_FIRST_AREA)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,first_area")
+endif()
+if (ECLAIR_FULL_DOC)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-full_doc=${ECLAIR_OUTPUT_DIR}/report_full_doc")
+endif()
+if (ECLAIR_FULL_ODT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-full_odt=${ECLAIR_OUTPUT_DIR}/report_full_odt")
+endif()
+
+message(STATUS "ECLAIR outputs have been written to: ${ECLAIR_OUTPUT_DIR}")
+message(STATUS "ECLAIR ECB files have been written to: ${ECLAIR_ANALYSIS_DATA_DIR}")
+
+add_custom_target(del_analysis_dir ALL
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${ECLAIR_ANALYSIS_DATA_DIR}
+  VERBATIM
+  USES_TERMINAL
+)
+
+# configure the camke script which will be used to replace the compiler call with the eclair_env
+# call which calls the compiler and to generate analysis files.
+configure_file(${CMAKE_CURRENT_LIST_DIR}/eclair.template ${ECLAIR_OUTPUT_DIR}/eclair.cmake @ONLY)
+
+set(launch_environment ${CMAKE_COMMAND} -P ${ECLAIR_OUTPUT_DIR}/eclair.cmake --)
+set(CMAKE_C_COMPILER_LAUNCHER ${launch_environment} CACHE INTERNAL "")
+
+# This target is used to generate the ECLAIR database when all the compilation is done and the
+# elf file was generated with this we cane make sure that the analysis is completed.
+add_custom_target(eclair_report ALL
+  COMMAND ${CMAKE_COMMAND} -E env
+    ECLAIR_DATA_DIR=${ECLAIR_ANALYSIS_DATA_DIR}
+    ECLAIR_OUTPUT_DIR=${ECLAIR_OUTPUT_DIR}
+    ECLAIR_PROJECT_ECD=${ECLAIR_PROJECT_ECD}
+    ${ECLAIR_REPORT} -quiet -eval_file=${ECLAIR_ECL_DIR}/db_generation.ecl
+  DEPENDS ${CMAKE_BINARY_DIR}/zephyr/zephyr.elf
+  VERBATIM
+  USES_TERMINAL
+  COMMAND_EXPAND_LISTS
+)
+
+# This command is used to generate the final reports from the database and print the overall results
+add_custom_target(eclair_summary_print ALL
+  COMMAND ${ECLAIR_REPORT}
+    -db=${ECLAIR_PROJECT_ECD} ${ECLAIR_REPORT_ADDITIONAL_OPTIONS}
+    -overall_txt=${ECLAIR_OUTPUT_DIR}/summary_overall.txt
+  COMMAND ${CMAKE_COMMAND} -E cat ${ECLAIR_OUTPUT_DIR}/summary_overall.txt
+  DEPENDS eclair_report
```

Comment:
`DEPENDS` is for files or outputs, not targets. You need to create a dependency with `add_dependencies`.

Reply:
ok it was a typo why add_dependencies didn't work